### DISCUSSION TITLE
⏳ 🔗 Loop Grid - "Get child of" query

### DISCUSSION THREAD

**Discussion by danielvigh** [elementor/elementor#21938] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Extend the Loop Grid query capabilities.

### Describe the solution you'd like

We would like to extend the Loop Grid query capabilities. For example: Get child of -> Current ID or Current page or Current post etc.

### Describe alternatives you've considered

JetEngine Listing Grid widget / Dynamic Posts widget by Dynamic.ooo

**Comment by ZoltanGalCMDesign** [comment]
Is there any news on this? I can not belive we can not list child pages in Elementor using Loop Grid...

**Comment by sergeilapp** [comment]
This functionality would be super helpful! I have needed these options multiple times now and the current need is to be able to Get child of -> select page

**Comment by danielvigh** [comment]
Is there any news on this?

**Comment by perks98** [comment]
Just had a project that needed this where I needed a list of child pages based on the current page. If you use the code below in a functions.php file of a child theme or use a code snippets related plugin; then under the 'Query' tab for the loop grid widget set source to Pages and set 'Query ID' to 'child_pages' and it will now show only child pages to the current page. This is just a quick temp solution until Elementor adds this natively.
```
/**
 * Update the Elementor query to use child pages for the current page. Use 'child_pages' under Query ID setting.
 *
 * @param \WP_Query $query The WordPress query instance.
 */
function child_pages_query_callback( $query ) {
    global $post;
    $query->set( 'post_parent', $post->ID );
}
add_action( 'elementor/query/child_pages', 'child_pages_query_callback' );
```

**Reply by ravenswoodmedia** [reply]
Thanks very much, works perfectly.

**Comment by ravenswoodmedia** [comment]
+1 for this. Should be a native function. @perks98 's solution works perfectly though.
